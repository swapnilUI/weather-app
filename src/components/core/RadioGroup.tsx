import React from 'react'
import Radio from '@material-ui/core/Radio'
import RadioGroup from '@material-ui/core/RadioGroup'
import FormControlLabel from '@material-ui/core/FormControlLabel'

interface IOptions {
    label:string
    value:string
}
interface IRadioGroupTagProps {
    onRadioChange:(selectedValue:string) => any
    options:IOptions[]
    defaultValue:string
}
export interface IRadioGroupTagState {
    value:string
}
export class RadioGroupTag extends React.PureComponent<IRadioGroupTagProps,IRadioGroupTagState>{
        constructor(props:IRadioGroupTagProps){
            super(props);
            this.state ={
                value:this.props.defaultValue
            }
        }
        public handleChange = (event:any) => {
            this.setState({
                value:event.target.value
            })
            this.props.onRadioChange(event.target.value);
        }
        public render(){
            return(
                <RadioGroup style={{
                    flexDirection:'row'
                }} value={this.state.value} onChange={this.handleChange}>
                    {
                        this.props.options.map( (btn:IOptions) => 
                            <FormControlLabel key={btn.value} value={btn.value} control={<Radio />} label={btn.label} />    
                        )
                    }
                </RadioGroup>
            )
        }
}