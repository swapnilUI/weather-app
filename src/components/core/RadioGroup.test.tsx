import React from 'react';
import { shallow } from 'enzyme';
import { RadioGroupTag, IRadioGroupTagState } from './RadioGroup';

const FakeOptions = [
    {
        label:'Celcius',
        value:'metric'
    },{
        label:'Fahrenheit',
        value:'imperial'
    }
];

const onRadioChange = jest.fn();

describe('MyComponent', () => {
  it('should render correctly', () => {
    const component = shallow(<RadioGroupTag onRadioChange={onRadioChange} defaultValue={FakeOptions[0].value} options={FakeOptions}  />);
    expect(component).toMatchSnapshot();
  });
  it('should check default value', () => {
    const component = shallow(<RadioGroupTag onRadioChange={onRadioChange} defaultValue={FakeOptions[0].value} options={FakeOptions}  />);
    const state:any = component.state()
    expect(state.value).toEqual(FakeOptions[0].value)
  });
  it('should change value on click', () =>{
    const component = shallow(<RadioGroupTag onRadioChange={onRadioChange} defaultValue={FakeOptions[0].value} options={FakeOptions}  />);
    const label = component.find(`label`).at(1);
    label.simulate('click');
    const state:any = component.state()
    expect(state.value).toEqual(FakeOptions[1].value)
  })
});