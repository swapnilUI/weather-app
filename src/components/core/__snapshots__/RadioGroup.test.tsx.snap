// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly 1`] = `ShallowWrapper {}`;
